parts/k8s/cloud-init/artifacts/cse_config.sh

```diff
@@ -487,6 +487,33 @@ configAddons() {
 }
 {{end}}
 {{- if HasNSeriesSKU}}
+
+installGPUDriversRun() {
+    {{- /* there is no file under the module folder, the installation failed, so clean up the dirty directory
+    when you upgrade the GPU driver version, please help check whether the retry installation issue is gone,
+    if yes please help remove the clean up logic here too */}}
+    set -x
+    MODULE_NAME="nvidia"
+    NVIDIA_DKMS_DIR="/var/lib/dkms/${MODULE_NAME}/${GPU_DV}"
+    KERNEL_NAME=$(uname -r)
+    if [ -d "${NVIDIA_DKMS_DIR}" ]; then
+        echo "the dkms module folder exists, we need to do the clean up first before retry to install."
+        if [ -x "$(command -v dkms)" ]; then
+          dkms remove -m ${MODULE_NAME} -v ${GPU_DV} -k ${KERNEL_NAME}
+        else
+          echo "the dkms tool is not found, so we just remove the directory to have a try"
+          rm -rf "${NVIDIA_DKMS_DIR}"
+        fi
+    fi
+    {{- /* we need to append the date to the end of the file because the retry will override the log file */}}
+    local log_file_name="/var/log/nvidia-installer-$(date +%s).log"
+    sh $GPU_DEST/nvidia-drivers-$GPU_DV --silent \
```

Comment:
Some of these arguments could use their short forms, specifically `--silent|-s`, `--kernel-name|-k`, and `--accept-license|-a`:

```shell
sh $GPU_DEST/nvidia-drivers-$GPU_DV -s \
        -k=${KERNEL_NAME} \
        --log-file-name=${log_file_name} \
        -a --no-drm --dkms --utility-prefix="${GPU_DEST}" --opengl-prefix="${GPU_DEST}"
exit $?
```

Sorry to nitpick, but we have to be careful to keep CSE content as lean as we can.

Reply:
updating it now.

Reply:
@mboersma updated. thanks.